test/unit/components/common/cgTable/cgTable_spec.js

```diff
@@ -0,0 +1,139 @@
+
+      parentScope.data = [
+        {
+          samples: 78,
+          negatives: 78,
+          positives: 0,
+          acronym: "ACC",
+          name: "adrenocortical cancer",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 390,
+          negatives: 390,
+          positives: 0,
+          acronym: "BLCA",
+          name: "bladder urothelial carcinoma",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 976,
+          negatives: 976,
+          positives: 0,
+          acronym: "BRCA",
+          name: "breast invasive carcinoma",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 193,
+          negatives: 193,
+          positives: 0,
+          acronym: "CESC",
+          name: "cervical & endocervical cancer",
+          isLoading: false,
+          isSelected: false
+        },
+        {
+          samples: 36,
+          negatives: 36,
+          positives: 0,
+          acronym: "CHOL",
+          name: "cholangiocarcinoma",
+          isLoading: false,
+          isSelected: true
+        }
+      ];
+
+      parentScope.columns = [
+        {
+          name: "Symbol",
+          id: "symbol",
+          isSortable: true
+        },
+        {
+          name: "Name",
+          id: "name",
+          isSortable: true
+        },
+        {
+          name: "Entrez ID",
+          id: "entrezgene",
+          isSortable: true
+        },
+        {
+          name: "Score",
+          id: "score",
+          isSortable: true
+        }
+      ];
+
+      element = angular.element(`
+          <cg-table
+            data="data"
+            columns="columns"
+          ></cg-table>        
+        `);
+
+      $compile(element)(parentScope);
+
+      controller = element.controller("cgTable");
+
+      parentScope.$digest();
+    })
+  );
+
+  it("displays the correct amount of columns", () => {
+    let columnCells = angular.element(element[0].querySelectorAll("th"));
+    expect(columnCells.length).toEqual(parentScope.columns.length + 1);
+  });
+
+  it("displays the correct amount of rows", () => {
+    let tableRows = angular.element(element[0].querySelectorAll("tbody tr"));
+    expect(tableRows.length).toEqual(parentScope.data.length);
+  });
+
+  it("detects if all results are selected", () => {
+    expect(controller.areAllResultsSelected()).toBe(false);
+  });
+
+  it("selects all results", () => {
+    controller.selectAllResults(true);
+    const areResultsAllSelected = parentScope.data.every(
+      result => result.isSelected === true
+    );
+    expect(areResultsAllSelected).toBe(true);
+  });
+
+  it("unselects all results", () => {
+    controller.selectAllResults(false);
+    const areResultsAllSelected = parentScope.data.every(
+      result => result.isSelected === true
+    );
+    expect(areResultsAllSelected).toBe(false);
+  });
```

Comment:
There is a small logic error here, meaning that this isn't quite testing what we're trying to test.

Let's say we have five results, labeled `A`, `B`, `C`, `D`, and `E`.

The test for each of the `result.isSelected` lines (129) will be `false` (if all goes well in `selectAllResults(false)`), so the summary at `parentScope.data.every()` will give us, effectively:

> `areResultsAllSelected` = `false` && `false` && `false` && `false` && `false`

This is, indeed, `false`. But what if `selectAllResults(false)` is broken, and it stops doing its job after the first entry? Then we have the following as a possible result:

> `areResultsAllSelected` = `false` && `false` && `false` && `false` && `true`

This is still `false`.

What we really want is a test that mirrors the "selects all results" test in a different way. I think the test should be something like:

```javascript
    const areResultsAllNotSelected = parentScope.data.every(
      result => result.isSelected === false
    );
    expect(areResultsAllNotSelected).toBe(true);
```